nodejs/eks/cluster.ts

```diff
@@ -254,12 +270,43 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
         kubeconfig: kubeconfig,
         provider: provider,
         vpcCni: vpcCni,
-        instanceRole: instanceRole,
-        instanceProfile: instanceProfile,
+        instanceProfile: instanceProfile! || undefined,
         eksNodeAccess: eksNodeAccess,
     };
 }
 
+// Add the specified policies to the existing IAM role.
+export type Policies =  pulumi.Input<aws.ARN>[];
+export function addPoliciesToExistingRole(
+    name: string,
+    role: aws.iam.Role,
+    policies: Policies) {
+
+    let counter = 0;
+    for (const policy of policies) {
+        // Create RolePolicyAttachment without returning it.
+        const rpa = new aws.iam.RolePolicyAttachment(
+            `${name}-policy-${counter++}`,
+            {
+                policyArn: policy,
+                role: role,
+            },
+        );
+    }
+}
+
+// Enable access to the EKS cluster for worker nodes by creating
+// the instance role mapping for aws-auth.
+export function createInstanceRoleMapping(arn: pulumi.Input<aws.ARN>): RoleMapping {
```

Comment:
For exported function - should use `/** * **/` style docs so they get picked up by doc generators.